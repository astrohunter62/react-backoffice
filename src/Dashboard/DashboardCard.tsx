import React from "react";
import classNames from "classnames";
import {
  Card,
  CardContent,
  Typography,
  withStyles,
  Avatar,
} from "@material-ui/core";
import DisabledIcon from "@material-ui/icons/Lock";

const styles = (theme: any) => ({
  root: {
    cursor: "pointer",
  },
  content: {
    paddingBottom: `${theme.spacing(2)}px !important`,
  },
  avatar: {
    float: "left",
    marginRight: theme.spacing(2),
    backgroundColor: theme.palette.secondary.dark,
  },
  title: {
    marginBottom: 0,
  },
  disabled: {
    opacity: 0.75,
    pointerEvents: "none",
    filter: "grayscale(30%)",
  },
  icon: {
    float: "right",
    opacity: 0.5,
  },
});

type DashboardCardProps = {
  title: string;
  description?: string;
  handleClick: (...args: any[]) => any;
  icon?: (...args: any[]) => any;
  isDisabled?: boolean;
  classes: {
    [key: string]: string;
  };
};

const DashboardCard: React.SFC<DashboardCardProps> = ({
  title,
  description,
  handleClick,
  icon: Icon,
  isDisabled,
  classes,
}) => {
  const rootClasses = classNames({
    [classes.root]: true,
    [classes.disabled]: isDisabled,
  });

  return (
    <Card onClick={handleClick} className={rootClasses}>
      <CardContent className={classes.content}>
        {isDisabled ? <DisabledIcon className={classes.icon} /> : null}

        {Icon ? (
          <Avatar className={classes.avatar}>
            <Icon />
          </Avatar>
        ) : null}

        <Typography variant="h5" className={classes.title} component="h2">
          {title}
        </Typography>

        {description ? (
          <Typography variant="body2">{description}</Typography>
        ) : null}
      </CardContent>
    </Card>
  );
};
DashboardCard.defaultProps = {
  isDisabled: false,
};

export default withStyles(styles as any)(DashboardCard);
