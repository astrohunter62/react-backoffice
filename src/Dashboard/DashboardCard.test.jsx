import React from 'react'
import renderer from 'react-test-renderer'

import DashboardCard from './DashboardCard'

it('renders correctly', () => {
  const tree = renderer
    .create(<DashboardCard
      title="Title"
      description="desc"
      handleClick={() => {}}
    />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
