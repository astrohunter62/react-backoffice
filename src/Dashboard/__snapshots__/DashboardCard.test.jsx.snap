// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-7 MuiPaper-elevation2-11 MuiPaper-rounded-8 DashboardCard-root-1"
  onClick={[Function]}
>
  <div
    className="MuiCardContent-root-34 DashboardCard-content-2"
  >
    <h2
      className="MuiTypography-root-35 MuiTypography-headline-40 DashboardCard-title-4"
    >
      Title
    </h2>
    <p
      className="MuiTypography-root-35 MuiTypography-body1-44"
    >
      desc
    </p>
  </div>
</div>
`;

exports[`renders disabled 1`] = `
<div
  className="MuiPaper-root-7 MuiPaper-elevation2-11 MuiPaper-rounded-8 DashboardCard-root-1 DashboardCard-disabled-5"
  onClick={[Function]}
>
  <div
    className="MuiCardContent-root-34 DashboardCard-content-2"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-59 DashboardCard-icon-6"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
      />
    </svg>
    <h2
      className="MuiTypography-root-35 MuiTypography-headline-40 DashboardCard-title-4"
    >
      Title
    </h2>
    <p
      className="MuiTypography-root-35 MuiTypography-body1-44"
    >
      desc
    </p>
  </div>
</div>
`;
